pkg/kubelet/kubelet.go

```diff
@@ -1556,11 +1561,43 @@ func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *api.ObjectFieldSelector, pod
 	}
 	switch internalFieldPath {
 	case "status.podIP":
-		return pod.Status.PodIP, nil
+		return kl.getPodIP(pod)
 	}
 	return fieldpath.ExtractFieldPathAsString(pod, internalFieldPath)
 }
 
+// getPodIP lazily finds the pod IP by consulting the following sources in order:
+//
+// 1.  The status of the passed in pod on the pod itself
```

Comment:
The thing is, if you don't pass the pod IP explicitly, you can't just strike that... unless you're willing to pay the price to query the container runtime every time because status manager wan't have the latest status if you just created the infra container in the same sync iteration.